stdlib/regex.gr

```diff
@@ -387,7 +395,7 @@ enum ParsedRegularExpression {
   RELookbehind(ParsedRegularExpression, Bool, Box<Number>, Box<Number>, Number, Number), // regex, is-match, lb-min, lb-max, n-start, num-n (lb-xx values patched in later)
   RECut(ParsedRegularExpression, Number, Number, Bool), // regex, n-start, num-n, needs-backtrack
   REReference(Number, Bool), // n, case-sensitive
```

Comment:
Question for the team: Would we want things like `IsMatch` and `CaseSensitive` to be type aliases?

Reply:
I think it'd be helpful!

Reply:
I disagree. IMO, the primary point of type aliases is to make certain values more clearly non-fungible with others. For something like `type RERange = List<(Number, Number)>`, it makes total sense to use an alias, since we want these to be a list with specific semantics. For `IsMatch` and `CaseSensitive`, there are no distinct semantics that these aliases are representing. In fact, I would go as far as to say that aliasing them is _harmful_, since now if I, as a user, see an `IsMatch` type in a signature somewhere, I have to go figure out what that is, only to discover that it is a boolean. The alias sort of seems redundant to me.